File: src/main/java/com/example/hospitalsystem/service/PacienteService.java
package com.example.hospitalsystem.service;

import com.example.hospitalsystem.model.AntecedenteMedico;
import com.example.hospitalsystem.model.HistoriaClinica;
import com.example.hospitalsystem.model.Paciente;
import com.example.hospitalsystem.repository.AntecedenteMedicoRepository;
import com.example.hospitalsystem.repository.HistoriaClinicaRepository;
import com.example.hospitalsystem.repository.PacienteRepository;
import com.example.hospitalsystem.security.Auditable;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

@Service
public class PacienteService {

    @Autowired
    private PacienteRepository pacienteRepository;

    @Autowired
    private HistoriaClinicaRepository historiaClinicaRepository;

    @Autowired
    private AntecedenteMedicoRepository antecedenteMedicoRepository;

    public List<Paciente> getAllPacientes() {
        return pacienteRepository.findAll();
    }

    public Optional<Paciente> getPacienteById(Long id) {
        return pacienteRepository.findById(id);
    }

    public Optional<Paciente> getPacienteByDni(String dni) {
        return pacienteRepository.findByDni(dni);
    }

    @Auditable(accion = "CREATE", entidad = "Paciente")
    @Transactional
    public Paciente createPaciente(Paciente paciente) {
        Paciente savedPaciente = pacienteRepository.save(paciente);
        HistoriaClinica historia = new HistoriaClinica();
        historia.setIdPaciente(savedPaciente.getIdPaciente());
        historia.setFechaApertura(LocalDate.now());
        historia.setObservaciones("Historia clínica creada automáticamente");
        historiaClinicaRepository.save(historia);
        return savedPaciente;
    }

    @Auditable(accion = "UPDATE", entidad = "Paciente")
    public Paciente updatePaciente(Long id, Paciente paciente) {
        if (pacienteRepository.existsById(id)) {
            paciente.setIdPaciente(id);
            return pacienteRepository.save(paciente);
        }
        return null;
    }

    @Auditable(accion = "DELETE", entidad = "Paciente")
    public boolean deletePaciente(Long id) {
        if (pacienteRepository.existsById(id)) {
            pacienteRepository.deleteById(id);
            return true;
        }
        return false;
    }

    public Optional<HistoriaClinica> getHistoriaClinicaByPaciente(Long idPaciente) {
        return historiaClinicaRepository.findByIdPaciente(idPaciente);
    }

    public List<AntecedenteMedico> getAntecedentesByHistoria(Long idHistoria) {
        return antecedenteMedicoRepository.findByIdHistoria(idHistoria);
    }

    @Auditable(accion = "CREATE", entidad = "AntecedenteMedico")
    public AntecedenteMedico addAntecedenteMedico(AntecedenteMedico antecedente) {
        return antecedenteMedicoRepository.save(antecedente);
    }
}

File: hospital_system/src/services/pacienteService.ts
import api from './api';

const pacienteService = {
    // ⭐ Métodos con nombres consistentes
    getAllPacientes: async () => {
        return await api.get('/pacientes');
    },

    getPacienteById: async (id: number) => {
        return await api.get(`/pacientes/${id}`);
    },

    getPacienteByDni: async (dni: string) => {
        return await api.get(`/pacientes/dni/${dni}`);
    },

    createPaciente: async (paciente: any) => {
        return await api.post('/pacientes', paciente);
    },

    updatePaciente: async (id: number, paciente: any) => {
        return await api.put(`/pacientes/${id}`, paciente);
    },

    deletePaciente: async (id: number) => {
        return await api.delete(`/pacientes/${id}`);
    },

    // Historia Clínica
    getHistoriaClinica: async (idPaciente: number) => {
        return await api.get(`/pacientes/${idPaciente}/historia-clinica`);
    },

    // Antecedentes Médicos
    getAntecedentes: async (idHistoria: number) => {
        return await api.get(`/pacientes/historia/${idHistoria}/antecedentes`);
    },

    addAntecedente: async (idHistoria: number, antecedente: any) => {
        return await api.post(`/pacientes/historia/${idHistoria}/antecedentes`, antecedente);
    },

    // ⭐ ALIAS para compatibilidad (mantener código antiguo funcionando)
    getAll: async () => pacienteService.getAllPacientes(),
    getById: async (id: number) => pacienteService.getPacienteById(id),
    getByDni: async (dni: string) => pacienteService.getPacienteByDni(dni),
    create: async (paciente: any) => pacienteService.createPaciente(paciente),
    update: async (id: number, paciente: any) => pacienteService.updatePaciente(id, paciente),
    delete: async (id: number) => pacienteService.deletePaciente(id),
};

export default pacienteService;
